services/QuillLMS/client/app/bundles/Teacher/containers/PremiumPricingGuide.jsx

```diff
@@ -26,34 +29,99 @@ const subscribers = [
   // const onMobile = () => size.width <= MAX_VIEW_WIDTH_FOR_MOBILE
   //
-  const { diagnosticActivityCount, lessonsActivityCount, independentPracticeActivityCount, } = props
-  return (
-    <div>
-      <div className="container premium-page">
-        <div className="overview text-center">
-          <PremiumPricingMinisRow {...props} />
-          <PremiumFeaturesTable
-            diagnosticActivityCount={diagnosticActivityCount}
-            independentPracticeActivityCount={independentPracticeActivityCount}
-            lessonsActivityCount={lessonsActivityCount}
-          />
-        </div>
+  state = {
+    showPremiumConfirmationModal: false,
+    showPurchaseModal: false,
+    subscriptionType: null,
+    subscriptionStatus: null,
+    userIsSignedIn: !!Number(document.getElementById('current-user-id').getAttribute('content')),
+    isScrolled: false
+  };
+
+  hidePremiumConfirmationModal = () => {
+    this.setState({ showPremiumConfirmationModal: false, });
+  };
+
+  hidePurchaseModal = () => {
+    this.setState({ showPurchaseModal: false, subscriptionType: null, });
+  };
+
+  showPremiumConfirmationModal = () => {
+    this.setState({ showPremiumConfirmationModal: true, });
+  };
+
+  showPurchaseModal = () => {
+    this.setState({ showPurchaseModal: true, });
+  };
+
+  showPurchaseModalForSchoolPurchase = () => {
+    this.setState({ subscriptionType: 'School', }, () => this.setState({ showPurchaseModal: true, }));
+  };
 
-        <div className="features text-center">
-          <SchoolPremium />
-          <SubscriberLogos subscribers={subscribers} />
+  updateSubscriptionStatus = subscription => {
+    this.setState({ subscriptionStatus: subscription,
+      showPremiumConfirmationModal: true,
+      showPurchaseModal: false, });
+  };
+
+  renderPremiumConfirmationModal = () => {
+    const { showPremiumConfirmationModal, subscriptionStatus, } = this.state
+    if (!showPremiumConfirmationModal) { return }
+    return (<PremiumConfirmationModal
+      hideModal={this.hidePremiumConfirmationModal}
+      show={showPremiumConfirmationModal}
+      subscription={subscriptionStatus}
+    />)
+  }
+
+  renderPurchaseModal = () => {
+    const {
+      showPurchaseModal,
+      subscriptionType
+    } = this.state
+    const { lastFour, } = this.props
+    if (!showPurchaseModal) { return }
+    return (<PurchaseModal
+      hideModal={this.hidePurchaseModal}
+      lastFour={lastFour}
+      show={showPurchaseModal}
+      subscriptionType={subscriptionType}
+      updateSubscriptionStatus={this.updateSubscriptionStatus}
+    />)
+  }
+
+  render() {
+    const { diagnosticActivityCount, independentPracticeActivityCount, lessonsActivityCount } = this.props
+    const { userIsSignedIn } = this.state
+    return (
+      <div>
+        <div className="container premium-page">
+          {userIsSignedIn ? <PremiumBannerBuilder originPage="premium" showPurchaseModal={this.showPurchaseModal} /> : ''}
```

Comment:
you can just do `userIsSignedIn && <PremiumBannerBuilder ... />` rather than render an empty string